webapp/app/static/jsx/components/AddressSearchBar.jsx

```diff
@@ -0,0 +1,96 @@
+import axios from 'axios';
+import React, { useEffect, useState, useRef } from 'react';
+import { InputGroup, FormControl, Form, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import { isValid } from './utils';
+import isTornado from '../../data/istornado';
+
+export default function AddressSearchBar({ onSubmit, inputAddress, setInputAddress }) {
+
+    const inputEl = useRef(null);
+    const [invalid, setInvalid] = useState(false);
+    const [tornadoTooltip, setTornadoTooltip] = useState('');
+
+    useEffect(() => {
+        if (!setInputAddress || inputAddress === undefined) {
+            [inputAddress, setInputAddress] = useState('');
+        }
+    }, [])
+
+    useEffect(() => {
+        if (inputEl.current.value.length === 0 && inputAddress.length > 0) {
+            inputEl.current.value = inputAddress;
+        }
+    }, [inputAddress])
+
+    const submitInputAddress = addr => {
+
+        if (!isValid(addr)) {
+            setInvalid(true);
+            return;
+        }
+        setInvalid(false);
+        onSubmit(addr);
+    }
+
+    const onChangeInputAddress = e => {
+        const addr = e.target.value;
+        e.preventDefault();
+        setInputAddress(addr);
+        axios.get('/utils/istornado?address=' + addr).then(response => {
```

Comment:
what happens if this hangs? is there a timeout?

Reply:
hm. i think the app will just continue onwards (without any popover) there's not currently a timeout except the default timeout by the browser. do u think there's a better way?

Reply:
oh cool, if it continues onward, then life is good